SmartDeviceLink/SDLIAPSession.m

```diff
@@ -141,10 +258,26 @@ - (SDLStreamErrorHandler)streamErroredHandler {
     };
 }
 
+- (SDLStreamHasSpaceHandler)streamHasSpaceHandler {
+    __weak typeof(self) weakSelf = self;
+    
+    return ^(NSStream *stream) {
+        __strong typeof(weakSelf) strongSelf = weakSelf;
+        
+        if (strongSelf.isDataSession) {
+            NSError *sendErr = nil;
+            
+            if (![strongSelf sdl_dequeueAndWriteToOutputStream:&sendErr] && sendErr != nil) {
+                [strongSelf sdl_handleOutputStreamWriteError:sendErr];
+            }
+        }
+    };
+}
 
 #pragma mark - Lifecycle Destruction
 
 - (void)dealloc {
+    self.sendDataQueue = nil;
```

Comment:
Everything I've seen from Apple says that `dealloc` post-ARC is for disposing of non instance variable resources. ivars are certainly torn down after `dealloc`, but I don't know that setting these to nil would necessarily help, because the object itself would only be partially torn down, correct? I'm just not sure this helps. I haven't seen these connection transition issues at all, so if you have verifiable, repeatable methods for testing an EA XPC issue, it would be nice if you could check that (assuming it's not a long process) or to open those steps up.